src/wollok/lang.wlk

```diff
@@ -2193,19 +2193,29 @@ class String {
    *     "this,could,be,a,list".split(",")   
    *          ==> Answers ["this", "could", "be", "a", "list"]
    */
-  method split(expression) {
-    self.checkNotNull(expression, "split")
-    const result = []
-    var me = self.toString() + expression
-    var first = 0
-    (0..me.size() - 1).forEach { i =>
-      if (me.charAt(i) == expression) {
-        result.add(me.substring(first, i))
-        first = i + 1
-      }
-    }
-    return result
-  }
+ method split(expression) {
+  const text = self
+	text.checkNotNull(expression, "split")
+	const result = []
+        if (text.equals("") && expression.equals("")) return result
+	const size = text.size() - 1
+	const expressionSize = expression.size()
+	if (expression.equals("")) {
+		(0 .. size).forEach{ i =>
+			if (i != size) result.add(text.charAt(i)) else text = text.charAt(i)
+		}
```

Comment:
algunas cosas que no termino de entender:

- no debería ser `0 .. size - 1`?
- text apunta a self, no deberíamos reapuntar `text`
- yo hubiera pensado en usar directamente fold

```ẁollok
if (...)
   return (0 .. size - 1).fold([], { total, i => total.add(self.charAt(i)) })
```

- `result` de hecho no sería necesario acá ni en la línea 2200 que directamente retorne `[]`

Después habría que corregirles el indentado también, en Eclipse se complica, con VSC es más fácil, hay unas [convenciones de escritura](https://github.com/uqbar-project/wollok-language/wiki/Coding-Standards-for-writing-this-repository)

Reply:
Es decir, el `if (i != size)` me hace ruido, están recorriendo la lista o usando el último caso como excepción dentro de la  iteración